pkg/recipes/terraform/config/providers/aws.go

```diff
@@ -118,9 +121,17 @@ func fetchAWSCredentials(ctx context.Context, awsCredentialsProvider credentials
 		return nil, err
 	}
 
-	if credentials == nil || credentials.AccessKeyID == "" || credentials.SecretAccessKey == "" {
-		logger.Info("AWS credentials are not registered, skipping credentials configuration.")
-		return nil, nil
+	switch credentials.Kind {
+	case ucp_datamodel.AWSAccessKeyCredentialKind:
+		if credentials.AccessKeyCredential == nil || credentials.AccessKeyCredential.AccessKeyID == "" || credentials.AccessKeyCredential.SecretAccessKey == "" {
+			logger.Info("AWS AccessKey credentials are not registered, skipping credentials configuration.")
+			return nil, nil
+		}
+	case ucp_datamodel.AWSIRSACredentialKind:
+		if credentials.IRSACredential == nil || credentials.IRSACredential.RoleARN == "" {
```

Comment:
Maybe I'm reading it wrong.. Shouldn't this be `and` operator instead of `or`? It will likely panic if we try to check `RoleArn` for nil `IRSACredential`. Same for the check above for access keys. This is also something that's not specific to this PR but just went unnoticed in previous iterations.


```suggestion
		if credentials.IRSACredential == nil && credentials.IRSACredential.RoleARN == "" {
```